src/qlever/commands/ui.py

```diff
@@ -184,70 +185,56 @@ def execute(self, args) -> bool:
+            except Exception as e:
+                log.error(
+                    f"Failed to get {ui_db_file} from {args.ui_image} "
+                    f"({e})"
+                )
                 return False
 
         # Start the QLever UI.
         try:
-            if pull_latest_image:
-                log.info(f"Pulling image `{args.ui_image}` for QLever UI")
-                run_command(pull_cmd)
-            log.info(f"Starting new container with name `{args.ui_container}`")
+            log.debug(
+                f"Starting new container with name `{args.ui_container}`"
+            )
             run_command(start_ui_cmd)
-        except subprocess.CalledProcessError as e:
-            log.error(f"Failed to start the QLever UI ({e})")
+        except Exception as e:
+            log.error(f"Failed to start container `{args.ui_container}` ({e})")
             return False
 
         # Check if config file with name `ui_config_file` exists. If not, try
         # to obtain it via `get_config_cmd` and set it as default.
         if Path(ui_config_file).exists():
             log.info(f"Found config file `{ui_config_file}` and reusing it")
         else:
-            log.info(
-                f"Checking if config `{ui_config_name}` exists in "
-                f"image `{args.ui_image}` "
-            )
             try:
-                config_yaml = run_command(get_config_cmd, return_output=True)
                 log.info(
-                    f"Found it and setting URL for SPARQL endpoint to "
-                    f"{sparql_endpoint}"
+                    f"Get default config file `{ui_config_file}` from image "
+                    f"`{args.ui_image}` and set endpoint to `{sparql_endpoint}`"
                 )
+                config_yaml = run_command(get_config_cmd, return_output=True)
                 config_dict = yaml.safe_load(config_yaml)
                 config_dict["config"]["backend"]["isDefault"] = True
                 config_dict["config"]["backend"]["baseUrl"] = sparql_endpoint
                 config_dict["config"]["backend"]["sortKey"] = 1
                 config_yaml = dict_to_yaml(config_dict)
-                log.info(f"Writing config to `{ui_config_file}`")
                 with open(ui_config_file, "w") as f:
                     f.write(config_yaml)
             except Exception as e:
-                log.error(
-                    f"Neither found config file `{ui_config_file}` nor "
-                    f"could obtain default config for `{ui_config_name}`"
-                )
-                log.info("")
-                log.info(f"Error message: {e}")
-                log.info("")
-                log.info(
-                    "TODO: provide further instructions for this case "
-                    "(obtain `default` config file, edit it, rename it, "
-                    "then try again)"
-                )
+                log.error(f"Export failed ({e})")
```

Comment:
note: A more descriptive error for the case that the backend does not exist would be nice. We'd have to make assumptions about the used UI image for that, so it's ok to not do that.

Reply:
@Qup42 The problem is that many kinds of errors can occur in this stretch of code: error getting the config file, error parsing it, error modifying it, error writing it. We would have a separate `try ... except` block for each, but that becomes kind of verbose. Not sure what the best way to do this is. I have now split it into two parts.